go/vt/srvtopo/resilient_server.go

```diff
@@ -79,17 +79,11 @@ type ResilientServer struct {
 
 // NewResilientServer creates a new ResilientServer
 // based on the provided topo.Server.
-func NewResilientServer(ctx context.Context, base *topo.Server, counterPrefix string) *ResilientServer {
+func NewResilientServer(ctx context.Context, base *topo.Server, counts *stats.CountersWithSingleLabel) *ResilientServer {
 	if srvTopoCacheRefresh > srvTopoCacheTTL {
 		log.Fatalf("srv_topo_cache_refresh must be less than or equal to srv_topo_cache_ttl")
 	}
 
-	metric := ""
-	if counterPrefix != "" {
-		metric = counterPrefix + "Counts"
-	}
-	counts := stats.NewCountersWithSingleLabel(metric, "Resilient srvtopo server operations", "type")
```

Comment:
The dynamic naming here is what is core to the problem and if the same name is (accidentally) used twice, it would panic. Instead we now always pass in the counter and build it in `init()` functions in the appropriate places beforehand.